pkg/database/database.go

```diff
@@ -0,0 +1,57 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/kubearchive/kubearchive/pkg/models"
+	_ "github.com/lib/pq"
+)
+
+type DBInterface interface {
+	QueryResources(kind, group, version string) ([]models.Resource, error)
+}
+
+type Database struct {
+	db                *sql.DB
+	resourceTableName string
+}
+
+func NewDatabase() (*Database, error) {
+	dataSource, err := ConnectionStr()
+	if err != nil {
+		return nil, err
+	}
+	db, err := sql.Open("postgres", dataSource)
+	if err != nil {
+		return nil, err
+	}
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+	return &Database{db, "archive"}, nil
+}
+
+func (db *Database) QueryResources(kind, group, version string) ([]models.Resource, error) {
+	query := fmt.Sprintf( //nolint:gosec
+		"SELECT data FROM %s WHERE kind=$1 AND api_version=$2", db.resourceTableName) //nolint:gosec
+	apiVersion := fmt.Sprintf("%s/%s", group, version)
+	rows, err := db.db.Query(query, kind, apiVersion)
```

Comment:
I think we should use queryContext instead so that any timeouts or deadlines specified by ctx will be respected

```suggestion
	rows, err := db.db.QueryContext(ctx, query, kind, apiVersion)
```

Reply:
updated!